src/System.Net.Primitives/src/System/Net/IPAddressParser.cs

```diff
@@ -213,19 +213,20 @@ public static unsafe bool Ipv6StringToAddress(ReadOnlySpan<char> ipSpan, ushort*
                 long result = 0;
                 if (!string.IsNullOrEmpty(scopeId))
                 {
-                    if (scopeId.Length < 2)
+                    uint interfaceIndex = Interop.Sys.GetNativeIPInterfaceIndex(scopeId);
+                    if (interfaceIndex >= 0)
```

Comment:
The man page seems to indicate that a return value of 0 means there was an error.

Reply:
Oh, indeed!